app/src/androidTest/java/com/android/sample/ui/activityDetails/CommentItemTest.kt

```diff
@@ -147,41 +148,58 @@ class CommentItemTest {
   }
 
   @Test
-  fun replyIsPosted_whenReplyInputIsFilledAndPosted() {
-    var replyPosted = false
+  fun repliesAreDisplayedCorrectly() {
     composeTestRule.setContent {
       CommentItem(
           profileId = testProfileId,
           comment = testComment,
-          onReplyComment = { replyText, _ -> replyPosted = replyText.isNotBlank() },
+          onReplyComment = { _, _ -> },
           onDeleteComment = {})
     }
 
-    composeTestRule
-        .onNodeWithTag("ReplyButton_${testComment.uid}")
-        .performClick() // Use the test tag
-    composeTestRule
-        .onNodeWithTag("replyInputField_${testComment.uid}")
-        .assertIsDisplayed() // Also ensure to set a test tag for the input field
-    composeTestRule
-        .onNodeWithTag("replyInputField_${testComment.uid}")
-        .performTextInput("New reply")
-    composeTestRule
-        .onNodeWithTag("postReplyButton_${testComment.uid}")
-        .performClick() // Use the test tag
-    assert(replyPosted)
+    composeTestRule.onNodeWithText("John: This is a reply").assertIsDisplayed()
   }
 
   @Test
-  fun repliesAreDisplayedCorrectly() {
+  fun replyButton_isNotDisplayedForReplies() {
+    val replyComment = testComment.replies.first() // The reply to the original comment
     composeTestRule.setContent {
       CommentItem(
           profileId = testProfileId,
-          comment = testComment,
+          comment = replyComment, // Pass in the reply comment
           onReplyComment = { _, _ -> },
-          onDeleteComment = {})
+          onDeleteComment = {},
+          allowReplies = false // Disable replies for this test
+          )
     }
 
+    // Ensure that the reply button is not displayed for replies
+    composeTestRule.onNodeWithTag("ReplyButton_${replyComment.uid}").assertDoesNotExist()
+  }
+
+  @Test
+  fun replies_areCorrectlyNestedUnderComments() {
+    // Set the content with the comment that has a reply
+    composeTestRule.setContent {
+      CommentSection(
+          profileId = testProfileId,
+          comments = listOf(testComment), // Pass in the comment with its reply
+          onReplyComment = { _, _ -> },
+          onDeleteComment = {},
+          onAddComment = {})
+    }
+
+    // Ensure the original comment is displayed
+    composeTestRule.onNodeWithText("Amine: This is a comment").assertIsDisplayed()
+
+    // Ensure the reply is displayed
     composeTestRule.onNodeWithText("John: This is a reply").assertIsDisplayed()
+
+    // Check that the reply is well nested, ensuring its indented and positioned under its origin
+    // comment
+    composeTestRule
+        .onNodeWithText("John: This is a reply")
+        .assertIsDisplayed()
+        .assertLeftPositionInRootIsEqualTo(40.dp)
   }
```

Comment:
This test clearly checks the visibility of both the comment and its reply, and also ensures that the reply is correctly indented, which is critical for the visual hierarchy. It might be helpful to add a brief comment explaining the significance of the 40.dp assertion for future reference, as this would clarify why this specific value ensures correct nesting.